src/DeductionManager.cpp

```
@@ -0,0 +1,164 @@
+
+    float fullWeight = 0.0f, consumed = 0.0f;
+    opt_get_actual_full_weight(&state.tag_data, &fullWeight);
+    opt_get_consumed_weight(&state.tag_data, &consumed);
+    float remaining = fullWeight - consumed;
+    if (remaining < 0) remaining = 0;
+
+    // Clamp: don't deduct more than what's on the tag
+    float deduction = (pending > remaining) ? remaining : pending;
+
+    NFCWriteRequest req;
+    memset(&req, 0, sizeof(req));
```

Cppcheck (2.20.0)

[portability] Using memset() on union which contains a floating point number.

(memsetClassFloat)